provider/google.go

```diff
@@ -123,6 +118,7 @@ func NewGoogleProvider(project string, dryRun bool) (Provider, error) {
 
 	provider := &googleProvider{
 		project: project,
+		domain:  domain,
```

Comment:
just a thought, instead of passing a `string domain`, we could pass a custom object, `ZoneResolver`, which would given a `map[zoneID]zoneName` would 1. filter out irrelevant hosted zones  2. resolve the best fitted hosted for a given record dns name